workflow-templates/check-general-formatting-task.yml

```diff
@@ -61,7 +61,7 @@ jobs:
 
       - name: Download latest editorconfig-checker release binary package
         id: download
-        uses: MrOctopus/download-asset-action@1.0
+        uses: MrOctopus/download-asset-action@1.1
```

Comment:
Installation of editorconfig-checker via `go install` is now well supported.

This approach would allow us to manage the version of the tool dependency via the Go modules system and Dependabot, as suggested for shfmt (https://github.com/arduino/tooling-project-assets/pull/396#discussion_r1418210810) and gon (https://github.com/arduino/tooling-project-assets/issues/360#issuecomment-2411475816), as well as other Go-based tool dependencies (which could even include Task).